src/hypothesis/internal/reflection.py

```diff
@@ -282,22 +282,42 @@ def extract_lambda_source(f):
     source = LINE_CONTINUATION.sub(' ', source)
     source = WHITESPACE.sub(' ', source)
     source = source.strip()
+    assert 'lambda' in source
+
+    tree = None
 
     try:
         tree = ast.parse(source)
     except SyntaxError:
         for i in hrange(len(source) - 1, len('lambda'), -1):
             prefix = source[:i]
             if 'lambda' not in prefix:
-                return if_confused
+                break
             try:
                 tree = ast.parse(prefix)
                 source = prefix
                 break
             except SyntaxError:
                 continue
-        else:
-            return if_confused
+    if tree is None:
+        if source.startswith('@'):
+            # This will always eventually find a valid expression because we
```

Comment:
“because we the decorator”?

Is that “we” unintentional?

Reply:
Yes. Edited the sentence from its original form and accidentally a word.